test-data/unit/check-selftype.test

```diff
@@ -356,3 +356,13 @@ reveal_type(A().member)  # E: Revealed type is '__main__.A*'
 reveal_type(B().member)  # E: Revealed type is '__main__.B*'
 
 [builtins fixtures/property.pyi]
+
+[case testSelfTypeNotSelfType]
+# Friendlier error messages for common mistakes. See #2950
+class A:
+    def f(x: int) -> None: ...
+    def g(self: None) -> None: ...
+[out]
+main:3: error: Self argument missing for a non-static method (or an invalid type for self)
+main:4: error: Invalid type for self, or extra argument type in function annotation
```

Comment:
The 'extra argument type in function annotation' only really makes sense as a separate thing from 'Invalid type for self' for comment annotations. However, I'm not sure if we can reliably determine whether an annotation was represented using a comment. Maybe only give the latter part of the message when in Python 2 mode?